src/clinical/service-worker-thread/tasks.ts

```diff
@@ -190,18 +190,31 @@ const mapEntityDocuments = (
 // Submitted Data Search Results
 function FilterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
   const { donorIds, submitterDonorIds } = query;
+  const useQueriedDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
 
-  const useFilteredDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
-
-  const filteredDonors = useFilteredDonors
-    ? donors.filter(donor => {
+  const filteredDonors = donors
+    .filter(donor => {
+      if (useQueriedDonors) {
         const { donorId, submitterId } = donor;
         const stringId = `${donorId}`;
         const donorMatch = donorIds?.some(id => stringId.includes(id));
         const submitterMatch = submitterDonorIds?.some(id => submitterId.includes(id));
```

Comment:
the `donorIds` you're matching on are super sets of the `stringId`?

Reply:
`donorIds` is a set of queried donors but can include partial search terms, i.e. if we search for '262' we will get all donors where donorId includes 262

I should leave a comment here b/c this is how we get fuzzy search working for DonorIds, which are numbers and can't be searched in MongoDB